### ISSUE TITLE
MySQL server is not available... Waiting 5 seconds

### ISSUE THREAD

**Issue by joaoheytor** [zabbix/zabbix-docker#556]
Hello guys,
I'm opening this issue because I'm facing an issue related to the MySQL database (I found some similar issues but all of those are already closed and the answer is not clear... at least to me!).

Anyway,
I'm using Docker-Compose environment with some other containers already working... the part of the Zabbix is:
 **mysql_zabbix**:
    image: mysql:5.7.23
    container_name: mysql_zabbix
    restart: always
    environment:
        - MYSQL_ROOT_PASSWORD=<REMOVED>
        - MYSQL_DATABASE=zabbix
        - MYSQL_USER=zabbix
        - MYSQL_PASSWORD=<REMOVED>
        - TZ=America/Sao_Paulo
    ports:
        - "3355:3306"
    volumes:
        - ./volumes/mysql_zabbix/mysql:/var/lib/mysql

And my Zabbix backend:
**zabbix_server_backend**:
    image: zabbix/zabbix-server-mysql:ubuntu-4.4.6
    container_name: zabbix_server_backend
    restart: always
    ports:
        - "10050:10050"
        - "10051:10051"
    volumes:
        - ./volumes/zabbix/usr/lib/zabbix/alertscripts:/usr/lib/zabbix/alertscripts
        - ./volumes/zabbix/usr/lib/zabbix/externalscripts:/usr/lib/zabbix/externalscripts
        - ./volumes/zabbix/var/lib/zabbix/modules:/var/lib/zabbix/modules
        - ./volumes/zabbix/var/lib/zabbix/enc:/var/lib/zabbix/enc
        - ./volumes/zabbix/var/lib/zabbix/ssh_keys:/var/lib/zabbix/ssh_keys
        - ./volumes/zabbix/var/lib/zabbix/ssl/certs:/var/lib/zabbix/ssl/certs
        - ./volumes/zabbix/var/lib/zabbix/ssl/keys:/var/lib/zabbix/ssl/keys
        - ./volumes/zabbix/var/lib/zabbix/ssl/ssl_ca:/var/lib/zabbix/ssl/ssl_ca
        - ./volumes/zabbix/var/lib/zabbix/snmptraps:/var/lib/zabbix/snmptraps
        - ./volumes/zabbix/var/lib/zabbix/mibs:/var/lib/zabbix/mibs
        - ./volumes/zabbix/var/lib/zabbix/export:/var/lib/zabbix/export
    environment:
        DB_SERVER_HOST: "mysql_zabbix"
        DB_SERVER_PORT: 3355
        MYSQL_USER: zabbix
        MYSQL_PASSWORD: <REMOVED>
        MYSQL_DATABASE: zabbix    
    links:
        - mysql_zabbix
    depends_on:
        - mysql_zabbix

**zabbix_server_frontend**:
    image: zabbix/zabbix-web-apache-mysql:ubuntu-4.4.6
    container_name: zabbix_server_frontend
    restart: always
    ports:
        - "9001:80"
        - "9002:443"
    environment:
        VIRTUAL_PORT: 80
        VIRTUAL_HOST: <REMOVED>
        LETSENCRYPT_HOST: <REMOVED>
        LETSENCRYPT_EMAIL: <REMOVED>
        DB_SERVER_HOST: mysql_zabbix
        DB_SERVER_PORT: 3355
        ZBX_SERVER_HOST: "zabbix_server_backend"
        MYSQL_USER: zabbix
        MYSQL_PASSWORD: <REMOVED>
        MYSQL_DATABASE: zabbix
        PHP_TZ: "America/Sao_Paulo"
    links:
        - mysql_zabbix_new
        - zabbix_server_backend
    depends_on:
        - nginx-proxy
        - mysql_zabbix
        - zabbix_server_backend

Any help here?

**Comment by dotneft** [comment]
If you use the same network with MySQL and Zabbix server, you need to specify 3306 port. Do not it?

Also it would be great to see logs from Zabbix server container.

**Comment by joaoheytor** [comment]
Hey @dotneft ,
I'm using another port to the MySQL because we already have another service running in that port... So I connect use to my MySQL... :'(

Anyway, the output of the docker logs command is:
![image](https://user-images.githubusercontent.com/13548034/75937438-9b63be80-5e63-11ea-9bf3-91aafd53540e.png)

Nothing helpful... :'( 

Oh I tried to find the zabbix_server.log inside the container without success... :'(
![image](https://user-images.githubusercontent.com/13548034/75937547-fbf2fb80-5e63-11ea-85e6-6b94d457be38.png)

Any other tip here?

**Comment by dotneft** [comment]
you need to use one network between server and MySQL. If you connect to Zabbix server container which port is available 3306 or 3355 to MySQL container?

**Comment by joaoheytor** [comment]
@dotneft 3355

**Comment by cesium147** [comment]
On 4.2 container zabbix-proxy-mysql have old mysql client and library and get error 
**ERROR 2026 (HY000): SSL connection error: unknown error number**

mysql-server have version 8.0:
```
$ docker-compose exec mysql-server bash

root@5c24f7f00240:/# mysql --version
mysql  Ver 8.0.19 for Linux on x86_64 (MySQL Community Server - GPL)
```

But proxy-mysql have version 5.7:
```$ docker-compose exec zabbix-proxy-mysql bash
root@6139ee27517d:/var/lib/zabbix# mysql --version
mysql  Ver 14.14 Distrib 5.7.27, for Linux (x86_64) using  EditLine wrapper

```$ docker-compose exec zabbix-proxy-mysql bash
root@6139ee27517d:/var/lib/zabbix# mysql -h mysql-server -u zabbix
ERROR 2026 (HY000): SSL connection error: unknown error number

root@6139ee27517d:/var/lib/zabbix# mysqladmin ping -h mysql-server -u zabbix
mysqladmin: connect to server at 'mysql-server' failed
error: 'SSL connection error: unknown error number'
```

I use "**--skip-ssl**" option to by-pass SSL
```
root@6139ee27517d:/var/lib/zabbix# root@60e193c8ce72:/var/lib/zabbix# mysql --skip-ssl -h mysql-server -u zabbix -p
WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 1087
Server version: 8.0.19 MySQL Community Server - GPL

Copyright (c) 2000, 2019, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> ^DBye



root@6139ee27517d:/var/lib/zabbix# mysqladmin ping --skip-ssl -h mysql-server -u zabbix -p
WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
Enter password:
mysqld is alive
```
And make patch for docker-entrypoint.sh 
```
diff --git a/proxy-mysql/ubuntu/docker-entrypoint.sh b/proxy-mysql/ubuntu/docker-entrypoint.sh
index ee280342..0d47b325 100755
--- a/proxy-mysql/ubuntu/docker-entrypoint.sh
+++ b/proxy-mysql/ubuntu/docker-entrypoint.sh
@@ -329,7 +329,7 @@ check_db_connect_mysql() {

     WAIT_TIMEOUT=5

-    while [ ! "$(mysqladmin ping -h ${DB_SERVER_HOST} -P ${DB_SERVER_PORT} -u ${DB_SERVER_ROOT_USER} \
+    while [ ! "$(mysqladmin ping --skip-ssl -h ${DB_SERVER_HOST} -P ${DB_SERVER_PORT} -u ${DB_SERVER_ROOT_USER} \
                 --password="${DB_SERVER_ROOT_PASS}" --silent --connect_timeout=10)" ]; do
         echo "**** MySQL server is not available. Waiting $WAIT_TIMEOUT seconds..."
         sleep $WAIT_TIMEOUT
@@ -383,7 +383,7 @@ mysql_query() {
     query=$1
     local result=""

-    result=$(mysql --silent --skip-column-names -h ${DB_SERVER_HOST} -P ${DB_SERVER_PORT} \
+    result=$(mysql --skip-ssl --silent --skip-column-names -h ${DB_SERVER_HOST} -P ${DB_SERVER_PORT} \
              -u ${DB_SERVER_ROOT_USER} --password="${DB_SERVER_ROOT_PASS}" -e "$query")

     echo $result
```

BUT zabbix_proxy crash on connect to mysql-server:
<details> 
  <summary>$ docker-compose restart zabbix-proxy-mysql</summary>
<p>

```
zabbix-proxy-mysql_1       | ** Deploying Zabbix proxy with mysql database
zabbix-proxy-mysql_1       | ** Preparing the system
zabbix-proxy-mysql_1       | Preparing Zabbix proxy
zabbix-proxy-mysql_1       | ** Using MYSQL_USER variable from ENV
zabbix-proxy-mysql_1       | ** Using MYSQL_PASSWORD variable from ENV
zabbix-proxy-mysql_1       | ** Using MYSQL_ROOT_PASSWORD variable from ENV
zabbix-proxy-mysql_1       | ********************
zabbix-proxy-mysql_1       | * DB_SERVER_HOST: mysql-server
zabbix-proxy-mysql_1       | * DB_SERVER_PORT: 3306
zabbix-proxy-mysql_1       | * DB_SERVER_DBNAME: zabbix_proxy
zabbix-proxy-mysql_1       | ********************
zabbix-proxy-mysql_1       | WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
zabbix-proxy-mysql_1       | mysqladmin: [Warning] Using a password on the command line interface can be insecure.
zabbix-proxy-mysql_1       | ** Creating 'zabbix' user in MySQL database
zabbix-proxy-mysql_1       | WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
zabbix-proxy-mysql_1       | mysql: [Warning] Using a password on the command line interface can be insecure.
zabbix-proxy-mysql_1       | WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
zabbix-proxy-mysql_1       | mysql: [Warning] Using a password on the command line interface can be insecure.
zabbix-proxy-mysql_1       | WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
zabbix-proxy-mysql_1       | mysql: [Warning] Using a password on the command line interface can be insecure.
zabbix-proxy-mysql_1       | WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
zabbix-proxy-mysql_1       | mysql: [Warning] Using a password on the command line interface can be insecure.
zabbix-proxy-mysql_1       | ** Database 'zabbix_proxy' already exists. Please be careful with database COLLATE!
zabbix-proxy-mysql_1       | WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
zabbix-proxy-mysql_1       | mysql: [Warning] Using a password on the command line interface can be insecure.
zabbix-proxy-mysql_1       | ** Table 'zabbix_proxy.dbversion' already exists.
zabbix-proxy-mysql_1       | WARNING: --ssl is deprecated and will be removed in a future version. Use --ssl-mode instead.
zabbix-proxy-mysql_1       | mysql: [Warning] Using a password on the command line interface can be insecure.
zabbix-proxy-mysql_1       | ** Preparing Zabbix proxy configuration file
zabbix-proxy-mysql_1       | ** Updating '/etc/zabbix/zabbix_proxy.conf' parameter "ProxyMode": ''... removed
zabbix-proxy-mysql_1       | ** Updating '/etc/zabbix/zabbix_proxy.conf' parameter "Server": 'zabbix-server'... updated

=== CUT ===

zabbix-proxy-mysql_1       | ** Updating '/etc/zabbix/zabbix_proxy.conf' parameter "TLSPSKIdentity": ''... removed
zabbix-proxy-mysql_1       | ** Updating '/etc/zabbix/zabbix_proxy.conf' parameter "TLSPSKFile": ''... removed
zabbix-proxy-mysql_1       | ** Cleaning the system
zabbix-proxy-mysql_1       | ########################################################
zabbix-proxy-mysql_1       | ** Starting Zabbix proxy
zabbix-proxy-mysql_1       | Starting Zabbix Proxy (active) [zabbix-proxy-mysql]. Zabbix 4.2.8 (revision cb5d5b1).
zabbix-proxy-mysql_1       | Press Ctrl+C to exit.
zabbix-proxy-mysql_1       |
zabbix-proxy-mysql_1       |    133:20200323:172310.666 Starting Zabbix Proxy (active) [zabbix-proxy-mysql]. Zabbix 4.2.8 (revision cb5d5b1).
zabbix-proxy-mysql_1       |    133:20200323:172310.666 **** Enabled features ****
zabbix-proxy-mysql_1       |    133:20200323:172310.666 SNMP monitoring:       YES
zabbix-proxy-mysql_1       |    133:20200323:172310.666 IPMI monitoring:       YES
zabbix-proxy-mysql_1       |    133:20200323:172310.666 Web monitoring:        YES
zabbix-proxy-mysql_1       |    133:20200323:172310.666 VMware monitoring:     YES
zabbix-proxy-mysql_1       |    133:20200323:172310.666 ODBC:                  YES
zabbix-proxy-mysql_1       |    133:20200323:172310.666 SSH2 support:          YES
zabbix-proxy-mysql_1       |    133:20200323:172310.666 IPv6 support:          YES
zabbix-proxy-mysql_1       |    133:20200323:172310.666 TLS support:           YES
zabbix-proxy-mysql_1       |    133:20200323:172310.666 **************************
zabbix-proxy-mysql_1       |    133:20200323:172310.666 using configuration file: /etc/zabbix/zabbix_proxy.conf
zabbix-proxy-mysql_1       |    133:20200323:172310.670 [Z3001] connection to database 'zabbix_proxy' failed: [2026] SSL connection error: unknown error number
zabbix-proxy-mysql_1       |    133:20200323:172310.670 Cannot connect to the database. Exiting...
zabbix-docker_zabbix-proxy-mysql_1 exited with code 1
```
</p>
</details>

**Comment by cesium147** [comment]
quick-fix - rollback mysql version to 8.0.18:
```
$ docker-compose rm --stop --force mysql-server
$ rm -rfv ./zbx_env/var/lib/mysql*
$ cat docker-compose.override.yml
---
version: "3.5"
services:
  mysql-server:
    image: mysql:8.0.18
$ git pull && docker-compose pull && docker-compose up -d --build
```

**Comment by dotneft** [comment]
I'm sorry, but please use modern Zabbix versions, 4.2 is quite quite old and unsupported anymore.
